fog/view/load-test/src/main.rs

```diff
@@ -41,79 +35,173 @@ struct Config {
-    )
-    .expect("kex");
+    let private_key = FromRandom::from_random(&mut rand::thread_rng());
+    let rng = VersionedKexRng::try_from_kex_pubkey(&rng_record.pubkey, &private_key).expect("kex");
     let search_keys = rng.take(num_search_keys).collect::<Vec<Vec<u8>>>();
 
-    loop {
-        let _resp = fog_view_client
-            .request(0, 0, search_keys.clone())
-            .expect("request");
-        num_reqs.fetch_add(1, Ordering::SeqCst);
+    while !stop_requested.load(Ordering::SeqCst) {
+        let start = Instant::now();
+        let result = fog_view_client.request(0, 0, search_keys.clone());
+        let duration = Instant::now().duration_since(start);
+
+        let mut counters = counters.lock().unwrap();
+        counters.num_requests += 1;
+        counters.total_millis_latency += u64::try_from(duration.as_millis()).unwrap();
+        if result.is_err() {
+            counters.num_errors += 1;
+        }
     }
 }
 
 fn main() {
     let config = Config::parse();
     let logger = create_root_logger();
 
-    let account_key =
-        mc_util_keyfile::read_keyfile(config.keyfile).expect("Could not read private key file");
+    let stop_requested = Arc::new(AtomicBool::default());
+    let r = stop_requested.clone();
+
+    ctrlc::set_handler(move || {
+        r.store(true, Ordering::SeqCst);
+    })
+    .expect("Error setting Ctrl-C handler");
 
-    let num_reqs = Arc::new(AtomicU64::new(0));
+    let counters = Arc::new(Mutex::new(Counters::default()));
     for _ in 0..config.num_workers {
         let logger = logger.clone();
-        let account_key = account_key.clone();
         let num_search_keys = config.num_search_keys;
-        let num_reqs = num_reqs.clone();
+        let counters = counters.clone();
+        let stop_requested = stop_requested.clone();
         let uri = config.view_uri.clone();
         let retry_config = config.grpc_retry_config;
 
         thread::spawn(move || {
             worker_thread(
                 uri,
                 retry_config,
-                account_key,
                 num_search_keys,
-                num_reqs,
+                counters,
+                stop_requested,
                 logger,
             )
         });
     }
 
-    loop {
-        let num_reqs_before = num_reqs.load(Ordering::SeqCst);
+    let start_test = Instant::now();
+
+    let mut last_counters = Counters::default();
+    let mut last_display = Instant::now();
+    while !stop_requested.load(Ordering::SeqCst) {
         thread::sleep(Duration::from_secs(1));
-        let num_reqs_after = num_reqs.load(Ordering::SeqCst);
 
-        println!("requests per second: {}", num_reqs_after - num_reqs_before);
+        let current_counters = counters.lock().unwrap().clone();
+        if current_counters.num_requests > last_counters.num_requests {
+            let now = Instant::now();
+            let duration = now.duration_since(last_display);
```

Comment:
Nit: 
```suggestion
            let duration = last_display.elapsed();
```

Reply:
so here i think i want to keep it as it is, because `now` is being used in two places: at this line, and at `last_display = now` below. If we are calling `Instant::now()` twice here, then there will be a small difference between those two values which feels less clean to me